### PR TITLE
Unify skilling session plugins

### PR DESCRIPTION
**Pull request by Desetude**
This PR unifies the skill session portions of the woodcutting, fishing, agility and #1879 cooking plugins into one plugin. 
Currently, I am unsure what to do with the rest of the woodcutting plugin as it only contains the notifications for bird nests and it is strange to have a whole plugin for just one tiny feature.

### CODE REVIEW

**Comment by deathbeam** [comment]
I think the woodcutting plugin is fine. Later someone wanted to add there tree indicators so it will be probably extended in future.

**Comment by Desetude** [comment]
Now ready for review.

File: runelite-api/src/main/java/net/runelite/api/AgilityCourse.java [hunk 1]
```diff
@@ -22,19 +22,20 @@
...
  */
-package net.runelite.client.plugins.agility;
+package net.runelite.api;
 
-import java.util.HashMap;
-import java.util.Map;
 import lombok.AllArgsConstructor;
 import lombok.Getter;
 
+import java.util.HashMap;
```
Comment (deathbeam) [inline runelite-api/src/main/java/net/runelite/api/AgilityCourse.java]: Fix your imports https://github.com/runelite/runelite/wiki/Code-Conventions

File: runelite-api/src/main/java/net/runelite/api/AgilityCourse.java [hunk 2]
```diff
@@ -48,25 +49,23 @@
...
 		{
 			courseXps.put(course.lastObstacleXp, course);
 		}
 	}
 
-	public static Courses getCourse(int exp)
+	@Getter
+	private final double totalXp;
+	private final int lastObstacleXp;
+	@Getter
```
Comment (deathbeam) [inline runelite-api/src/main/java/net/runelite/api/AgilityCourse.java]: put the getters together

File: runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingPlugin.java [hunk 3]
```diff
@@ -50,9 +41,15 @@
 import net.runelite.client.plugins.PluginDependency;
 import net.runelite.client.plugins.PluginDescriptor;
 import net.runelite.client.plugins.xptracker.XpTrackerPlugin;
-import net.runelite.client.ui.overlay.Overlay;
 import net.runelite.client.util.QueryRunner;
 
+import javax.inject.Inject;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingPlugin.java]: Fix your imports https://github.com/runelite/runelite/wiki/Code-Conventions

File: runelite-client/src/main/java/net/runelite/client/plugins/skillsessions/SkillSessionsConfig.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+@ConfigGroup(
+		keyName = "skillsessions",
+		name = "Skill Session",
+		description = "Configuration for the skill sessions plugin"
+)
+public interface SkillSessionsConfig
+{
+	@ConfigItem(
+			keyName = "statTimeout",
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/skillsessions/SkillSessionsConfig.java]: Fix your indentation https://github.com/runelite/runelite/wiki/Code-Conventions (in Idea settings change continuation indent from 8 to 4)

File: runelite-client/src/main/java/net/runelite/client/plugins/skillsessions/SkillSessionsOverlay.java [hunk 5]
```diff
@@ -22,72 +22,43 @@
...
 import net.runelite.client.ui.overlay.Overlay;
 import net.runelite.client.ui.overlay.OverlayPosition;
-import net.runelite.client.ui.overlay.OverlayPriority;
 import net.runelite.client.ui.overlay.components.PanelComponent;
 
-public class LapCounterOverlay extends Overlay
-{
-	private final AgilityPlugin plugin;
-	private final AgilityConfig config;
+import java.awt.*;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/skillsessions/SkillSessionsOverlay.java]: Fix your imports https://github.com/runelite/runelite/wiki/Code-Conventions

File: runelite-client/src/main/java/net/runelite/client/plugins/skillsessions/SkillSessionsPlugin.java [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+import com.google.common.collect.ClassToInstanceMap;
+import com.google.common.collect.MutableClassToInstanceMap;
+import com.google.common.eventbus.Subscribe;
+import com.google.common.primitives.Ints;
+import com.google.inject.Inject;
+import com.google.inject.Provides;
+import lombok.Getter;
+import lombok.Setter;
+import net.runelite.api.*;
+import net.runelite.api.events.*;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/skillsessions/SkillSessionsPlugin.java]: Fix your imports https://github.com/runelite/runelite/wiki/Code-Conventions

File: runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingPlugin.java (hunk 3 above)
Comment (Desetude) [reply runelite-client/src/main/java/net/runelite/client/plugins/fishing/FishingPlugin.java]: Done.

**Comment by Adam-** [comment]
I don't think having all of the session code in one plugin is that right approach. Also the config being not with the respective plugins is probably also very confusing. I would expect each plugin to expose some class which implements SkillingSession

**Comment by deathbeam** [comment]
Each plugin having it's own timeout setting and toggle is very confusing imo as it is all essentially same thing and still scattered around.

**Pull request closed without merging**